extensions/objects/templates/image-editor.component.ts

```diff
@@ -982,6 +982,24 @@ export class ImageEditorComponent implements OnInit, OnChanges {
       let gifHeight = dimensions.height;
       this.processGIFImage(imageDataURI, gifWidth, gifHeight, null, successCb);
     } else if (mimeType === 'data:image/svg+xml') {
+      // Check if the SVG containes 'data-name' attribute and remove it.
+      // We are removing this attribute, because currently it is not in
+      // the allowlist of valid attributes. The allowlist is based on the
+      // list of tags and attributes specified in this project:
+      // https://github.com/cure53/DOMPurify
+      // Complete list of valid SVG attributes is present at
+      // 'assets/constants.ts'.
+      const svgString = atob(imageDataURI.split(',')[1]);
+      let domParser = new DOMParser();
+      let doc = domParser.parseFromString(svgString, 'image/svg+xml');
```

Comment:
Looks like we parse the SVG again in line 1004 to identify bad attrs / tags. Should we consider moving this logic to `SvgSanitizerService` and see if some code can be reused (e.g. we could specify a parameter that is a list of attrs / tags to be stripped)?

Reply:
The code has been updated now.